### PR TITLE
Pass the container name to deployment config log options

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/10186 (partially)

We are not passing the container name when we converting the deployment logs options to pod log options so in case you have pod with multiple container and you do `oc logs dc/foo -c bar` you will always get the error that you have to pick the container. This PR fixes that.

### CODE REVIEW

**Comment by mfojtik** [comment]
@soltysh @smarterclayton PTAL

**Comment by mfojtik** [comment]
[test]

File: pkg/cmd/cli/cmd/logs.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ func NewCmdLogs(name, parent string, f *clientcmd.Factory, out io.Writer) *cobra
 // resource a user requested to view its logs and creates the appropriate logOptions
 // object for it.
 func (o *OpenShiftLogsOptions) Complete(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	containerName := kcmdutil.GetFlagString(cmd, "container")
 	if err := o.KubeLogOptions.Complete(f.Factory, out, cmd, args); err != nil {
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/logs.go]: KubeLogOptions.Complete already pulls the container name out of the arg... should be available in `podLogOptions.Container`

**Comment by mfojtik** [comment]
@kargakis any objects if you are online?

File: pkg/cmd/cli/cmd/logs.go (hunk 1 above)
Comment (mfojtik) [inline pkg/cmd/cli/cmd/logs.go]: @liggitt I tried that and it is not for some reason. I can try harder :-)

Comment (mfojtik) [inline pkg/cmd/cli/cmd/logs.go]: @liggitt actually I think it was the other func that brreak it, you're right it is part of that.

**Comment by mfojtik** [comment]
@liggitt fixed.

**Comment by soltysh** [comment]
LGTM

**Comment by mfojtik** [comment]
[merge]

**Comment by smarterclayton** [comment]
Please add a test.

**Comment by mfojtik** [comment]
@smarterclayton will do as follow up

**Comment by smarterclayton** [comment]
Looks like you have a chance now

**Comment by kargakis** [comment]
LGTM

**Comment by kargakis** [comment]
```
FAILURE after 59.820s: test/cmd/deployments.sh:138: executing 'oc get pods -l openshift.io/deployer-pod-for.name=test-deployment-config-multicontainer-1 | wc -l' expecting any result and text '0'; re-trying every 0.2s until completion or 60.000s: the command timed out
Standard output from the command:
2
... repeated 90 times
```

**Comment by mfojtik** [comment]
@kargakis interesting, that means that the deployer pod is not going away?

**Comment by mfojtik** [comment]
(the test is failing because test-cmd does not have access to docker, I will rework it to extended test).

**Comment by mfojtik** [comment]
[merge]

**Comment by kargakis** [comment]
```
• Failure [45.577 seconds]
deploymentconfigs
/data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:648
  with logging when have multiple containers [Conformance]
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:459
    should be able to get the logs from each container [It]
    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:457

    Expected an error to have occurred.  Got:
        <nil>: nil

    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:451
------------------------------
```

**Comment by mfojtik** [comment]
@kargakis condition fixed, [merge]

**Comment by mfojtik** [comment]
[test]

File: test/extended/deployments/deployments.go [hunk 2]
```diff
@@ -458,6 +459,24 @@ var _ = g.Describe("deploymentconfigs", func() {
 		})
 	})
 
+	g.Describe("with logging when have multiple containers [Conformance]", func() {
+		g.AfterEach(func() {
+			failureTrap(oc, "hook", g.CurrentGinkgoTestDescription().Failed)
```
Comment (kargakis) [inline test/extended/deployments/deployments.go]: Should be "test-deployment-config-multicontainer"

File: test/extended/deployments/deployments.go [hunk 3]
```diff
@@ -458,6 +459,24 @@ var _ = g.Describe("deploymentconfigs", func() {
 		})
 	})
 
+	g.Describe("with logging when have multiple containers [Conformance]", func() {
```
Comment (kargakis) [inline test/extended/deployments/deployments.go]: "with multiple containers"

Comment (mfojtik) [inline test/extended/deployments/deployments.go]: fixed

**Pull request merged**